sdk/metricsadvisor/Azure.AI.MetricsAdvisor/perf/Tests/GetIncidentRootCausesTest.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.AI.MetricsAdvisor.Perf
+{
+    public class GetIncidentRootCausesTest : MetricsAdvisorTest<MetricsAdvisorPerfOptions>
+    {
+        public GetIncidentRootCausesTest(MetricsAdvisorPerfOptions options) : base(options)
+        {
+            ValidateOptions();
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            int count = 0;
+
+            foreach (var _ in Client.GetIncidentRootCauses(DetectionConfigurationId, IncidentId, cancellationToken: cancellationToken))
+            {
+                if (++count >= Options.Count)
+                {
+                    break;
+                }
+            }
```

Comment:
A more C# canonical way of writing the above block would be -
```
            for (int count = 0; count < Options.Count; count++)
            {
                Client.GetIncidentRootCauses(DetectionConfigurationId, IncidentId, cancellationToken: cancellationToken));
            }
```

Reply:
I've been using so many async enumerables that I forgot how to code normally.

Reply:
Actually, what you're suggesting makes multiple calls to `GetIncidentRootCauses`, which is not what I'm trying to do. I have a single call and I'm iterating over the first elements of the returned collection.

Reply:
You could pull the call out of the block and then iterate in the loop.
It is surprising to see that code, that's all.